gui/wxpython/gis_set.py

```diff
@@ -537,6 +540,9 @@ def SuggestDatabase(self):
                 'your home directory. '
                 'Press Browse button to select the directory.'))
 
+        self.SetLocation(grassdatabase, location, mapset_name)
+        self.ExitSuccessfully()
+
```

Comment:
Having this call at the end of function called `SuggestDatabase` should be suspicious.

The point is that the first time user* will be brought into the newly created location, not everybody.

On top of that, just from reading the code, it looks like `grassdatabase, location, mapset_name` will not be defined on like 543 unless the `if path is None:` case happens. Did you test this for case when `path` is not `None`, i.e., `get_possible_database_path()` found something?

* Here defined as the case when the startup location was created.

Reply:
The call is now at the right place in terms of starting there for the first-time user.